packages/react/src/hooks/useSubscriptionScopes.ts

```diff
@@ -1,80 +1,35 @@
-import type { NotifyClientTypes } from "@walletconnect/notify-client";
-import { useCallback, useEffect, useState } from "react";
-import { ErrorOf, HooksReturn, LoadingOf, SuccessOf } from "../types/hooks";
+import { useEffect, useState } from "react";
+import { HooksReturn, SuccessOf } from "../types/hooks";
 import { useWeb3InboxClient } from "./useWeb3InboxClient";
 import { useSubscriptionState } from "../utils/snapshots";
+import { Web3InboxClient } from "@web3inbox/core";
 
-type SubscriptionScopesReturn = HooksReturn<
-  { scopes: NotifyClientTypes.ScopeMap },
-  { updateScopes: (scope: string[]) => Promise<boolean> },
-  "updateScopes"
+type GetNotificationTypesReturnType = Awaited<
+  ReturnType<Web3InboxClient["getNotificationTypes"]>
 >;
+type UseSubscriptionScopesData = GetNotificationTypesReturnType;
+type UseSubscriptionScopesReturn = HooksReturn<UseSubscriptionScopesData>;
 
 export const useSubscriptionScopes = (
   account?: string,
   domain?: string
-): SubscriptionScopesReturn => {
-  const { data: w3iClient, error: clientError } = useWeb3InboxClient();
+): UseSubscriptionScopesReturn => {
+  const { data: w3iClient } = useWeb3InboxClient();
+
   const { subscriptions: subscriptionsTrigger } = useSubscriptionState();
-  const [subScopes, setSubScopes] = useState<NotifyClientTypes.ScopeMap>(
+  const [data, setData] = useState<UseSubscriptionScopesData>(
     w3iClient?.getNotificationTypes(account) ?? {}
   );
 
-  const [error, setError] = useState<string | null>(null);
-
   useEffect(() => {
     if (w3iClient) {
-      setSubScopes(w3iClient.getNotificationTypes(account, domain));
+      setData(w3iClient.getNotificationTypes(account, domain));
     }
   }, [w3iClient, account, subscriptionsTrigger, domain]);
 
-  const updateScopes = useCallback(
-    (scope: string[]) => {
-      if (w3iClient) {
-        return w3iClient.update(scope, account, domain);
-      } else {
-        setError(
-          "Trying to update scope before Web3Inbox Client was initialized "
-        );
-        return Promise.resolve(false);
-      }
-    },
-    [w3iClient, account, domain]
-  );
-
-  if (!w3iClient) {
-    return {
-      data: null,
-      error: null,
-      isLoading: true,
-      updateScopes,
-    } as LoadingOf<SubscriptionScopesReturn>;
-  }
-
-  if (clientError) {
-    return {
-      data: null,
-      error: {
-        client: clientError.client,
-      },
-      isLoading: false,
-      updateScopes,
-    } as ErrorOf<SubscriptionScopesReturn>;
-  }
-
-  if (error) {
-    return {
-      data: null,
-      error: { updateScopes: { message: error } },
-      isLoading: false,
-      updateScopes,
-    } as ErrorOf<SubscriptionScopesReturn>;
-  }
-
   return {
-    data: { scopes: subScopes },
+    data,
     error: null,
     isLoading: false,
-    updateScopes,
```

Comment:
The `data` is the subscription scopes object in `useSubscriptionScopes` while the `data` is response of the `update` method in `useUpdateSubscription`.  

Since they both have `data/loading/error` states, I've separated them. Do you think we should move the `update` method to `useSubscriptionScopes`? 

Also while the `update` method is not only for updating the notification types, the `useSubscriptionScopes` is only fetching the notification types. I though we could even update the `useSubscriptionScope` as `useNotificationTypes`. The fetcher method name is `getNotificationTypes` on the client

Reply:
i think we should combine them into `useNotificationTypes` with `update`